ckanext/datastore/helpers.py

```diff
@@ -67,6 +67,24 @@ def should_fts_index_field_type(field_type):
     return field_type.lower() in ['tsvector', 'text', 'number']
 
 
+def get_function_names_from_sql(context, sql):
+
+    def _get_function_names(tokens, function_names):
+        for token in tokens:
+            if isinstance(token, sqlparse.sql.Function):
+                function_name = token.get_name()
+                if function_name not in function_names:
+                    function_names.append(function_name)
+            if hasattr(token, 'tokens'):
+                return _get_function_names(token.tokens, function_names)
+
+        return function_names
+
+    parsed = sqlparse.parse(sql)[0]
+
+    return _get_function_names(parsed.tokens, [])
```

Comment:
I'm seeing duplicated logic in the lines below
https://github.com/OpenGov-OpenData/ckan/blob/ff034c374376d7fecadaf74f36887ac4785334fa/ckanext/datastore/helpers.py#L151-L166

Reply:
removed